test/types/community/collection/findX.test-d.ts

```diff
@@ -102,6 +102,23 @@ expectType<{ cost: number } | undefined>(
   await collectionBag.findOne<{ cost: number }>({ color: 'red' }, { projection: { cost: 1 } })
 );
 
+// NODE-3468 The generic in find and findOne no longer affect the filter type
+// Could not find a way to use Parameters<pets.find<Type>> to check the actual filter argument type
+type Pet = { type: string; age: number };
+const pets = db.collection<Pet>('pets');
+
+expectType<{ crazy: number }[]>(
+  await pets
+    .find<{ crazy: number }>({ type: 'dog', age: 1 })
+    .toArray()
+);
+
+expectType<{ 'dot.keys': number }[]>(
```

Comment:
I think we can remove this example and the corresponding comment in L106; however, I think it's worth adding a dot key case to the filterQuery  test file to show that it is assignable even when the referenced key is the wrong type

Reply:
Added a test to the bottom of filterQuery test file